store/PoolInstancesSlice/index.tsx

```diff
@@ -320,6 +320,29 @@ export const createPoolsInstancesSlice: StateSlice<IPoolsInstancesSlice> = (set,
                 });
         });
     },
+    simulateUpdateAvgEntryPrices: (pool_) => {
+        set((state) => {
+            const pool = state.pools[pool_];
+            if (!pool) {
+                return state;
+            }
+            const {
+                avgLongEntryPriceAggregate,
+                avgLongEntryPriceWallet,
+                avgShortEntryPriceWallet,
+                avgShortEntryPriceAggregate,
+            } = pool.userBalances.tradeStats;
+            const newAvgLongPriceWallet = (avgLongEntryPriceAggregate.plus(avgLongEntryPriceWallet)).div(2);
+            const newAvgShortPriceWallet = (avgShortEntryPriceAggregate.plus(avgShortEntryPriceWallet)).div(2);
+            state.pools[pool_].userBalances.tradeStats = {
+                ...state.pools[pool_].userBalances.tradeStats,
+                avgLongEntryPriceAggregate: new BigNumber(0),
+                avgShortEntryPriceAggregate: new BigNumber(0),
+                avgLongEntryPriceWallet: newAvgLongPriceWallet,
+                avgShortEntryPriceWallet: newAvgShortPriceWallet,
+            };
```

Comment:
is this supposed to mutate or return a new state? noticed the early return statement returns an object

Reply:
yeah I dont think I actually need to `return state` in that above one but did it just in case. Immer set returns whatever state is at the end of the function the set